epochStart/bootstrap/process.go

```diff
@@ -264,7 +265,10 @@ func (e *epochStartBootstrap) prepareEpochZero() (Parameters, error) {
 		return Parameters{}, err
 	}
 
-	shardIDToReturn := e.applyShardIDAsObserverIfNeeded(e.genesisShardCoordinator.SelfId())
+	shardIDToReturn := e.genesisShardCoordinator.SelfId()
+	if !e.isNodeInGenesisNodesConfig() {
```

Comment:
can you add a unit test for this :) ? good find.

Reply:
done. added 2 unit tests